crates/typst/src/text/font/color.rs

```diff
@@ -33,15 +35,88 @@ pub fn frame_for_glyph(font: &Font, glyph_id: u16) -> Frame {
 
     if let Some(raster_image) = ttf.glyph_raster_image(glyph_id, u16::MAX) {
         draw_raster_glyph(&mut frame, font, upem, raster_image);
+    } else if ttf.is_color_glyph(glyph_id) {
+        draw_colr_glyph(&mut frame, upem, ttf, glyph_id);
     } else if ttf.glyph_svg_image(glyph_id).is_some() {
         draw_svg_glyph(&mut frame, upem, font, glyph_id);
-    } else if ttf.is_color_glyph(glyph_id) {
-        draw_colr_glyph(&mut frame, font, glyph_id);
     }
 
     frame
 }
 
+fn draw_colr_glyph(
+    frame: &mut Frame,
+    upem: Abs,
+    ttf: &ttf_parser::Face,
+    glyph_id: GlyphId,
+) -> Option<()> {
+    let mut svg = XmlWriter::new(xmlwriter::Options::default());
+
+    let width = ttf.global_bounding_box().width() as f64;
+    let height = ttf.global_bounding_box().height() as f64;
+    let x_min = ttf.global_bounding_box().x_min as f64;
+    let y_max = ttf.global_bounding_box().y_max as f64;
+    let tx = -x_min;
+    let ty = -y_max;
+
+    svg.start_element("svg");
+    svg.write_attribute("xmlns", "http://www.w3.org/2000/svg");
+    svg.write_attribute("xmlns:xlink", "http://www.w3.org/1999/xlink");
+    svg.write_attribute("width", &width);
+    svg.write_attribute("height", &height);
+    svg.write_attribute_fmt("viewBox", format_args!("0 0 {width} {height}"));
+
+    let mut path_buf = String::with_capacity(256);
+    let gradient_index = 1;
+    let clip_path_index = 1;
+
+    svg.start_element("g");
+    svg.write_attribute_fmt(
+        "transform",
+        format_args!("matrix(1 0 0 -1 0 0) matrix(1 0 0 1 {tx} {ty})"),
+    );
+
+    let mut glyph_painter = GlyphPainter {
+        face: ttf,
+        svg: &mut svg,
+        path_buf: &mut path_buf,
+        gradient_index,
+        clip_path_index,
+        palette_index: 0,
+        transform: ttf_parser::Transform::default(),
+        outline_transform: ttf_parser::Transform::default(),
+        transforms_stack: vec![ttf_parser::Transform::default()],
+    };
+
+    ttf.paint_color_glyph(glyph_id, 0, RgbaColor::new(0, 0, 0, 255), &mut glyph_painter)
+        .unwrap();
+    svg.end_element();
+
+    let mut data = svg.end_document().into_bytes();
+
+    let mut decoded = vec![];
+    if data.starts_with(&[0x1f, 0x8b]) {
+        let mut decoder = flate2::read::GzDecoder::new(data.as_bytes());
+        decoder.read_to_end(&mut decoded).ok()?;
+        data = decoded;
+    }
```

Comment:
This seems unnecessary since you've just encoded the document and it's surely not compressed.

Reply:
Good point, I copy-pasted too much. 😅